gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactor.java

```diff
@@ -0,0 +1,314 @@
+
+    } finally {
+      this.executorService.shutdown();
+      try {
+        executorService.awaitTermination(Long.MAX_VALUE, TimeUnit.SECONDS);
+        for (Future<?> future : this.futures) {
+          future.get();
+        }
+      } catch (InterruptedException e) {
+        LOG.warn("Interrupted while waiting for Hadoop jobs to complete", e);
+        throw new RuntimeException(e);
+      } catch (ExecutionException e) {
+        LOG.warn("Got an Exception while processing an input folder", e);
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  /**
+   * Each subdir name in this.inputDir is considered a topic, if it satisfies
+   * blacklist and whitelist.
+   */
+  @SuppressWarnings("deprecation")
+  private Set<String> findAllTopics() throws IOException {
+    Set<String> topics = Sets.newHashSet();
+    List<Pattern> blacklist = getBlackList();
+    List<Pattern> whitelist = getWhiteList();
+    for (FileStatus status : this.fs.listStatus(new Path(this.inputDir))) {
+      if (status.isDir()) {
+        String topic = status.getPath().getName();
+        if (survived(topic, blacklist, whitelist)) {
+          LOG.info("found topic: " + status.getPath().getName());
+          topics.add(status.getPath().getName());
+        }
+      }
+    }
+    return topics;
+  }
+
+  private List<Pattern> getBlackList() {
+    List<String> list = this.state.getPropAsList(ConfigurationKeys.COMPACTION_BLACKLIST, StringUtils.EMPTY);
+    return getPatternsFromStrings(list);
+  }
+
+  private List<Pattern> getWhiteList() {
+    List<String> list = this.state.getPropAsList(ConfigurationKeys.COMPACTION_WHITELIST, StringUtils.EMPTY);
+    return getPatternsFromStrings(list);
+  }
+
+  private static List<Pattern> getPatternsFromStrings(List<String> strings) {
+    List<Pattern> patterns = Lists.newArrayList();
+    for (String s : strings) {
+      patterns.add(Pattern.compile(s, Pattern.CASE_INSENSITIVE));
+    }
+    return patterns;
+  }
+
+  /**
+   * If whitelist is non-empty, a topic survives if it matches the whitelist.
+   * Otherwise, a topic survives if it doesn't match the blacklist.
+   * Whitelist and blacklist use regex patterns (NOT glob patterns).
+   */
+  private static boolean survived(String topic, List<Pattern> blacklist, List<Pattern> whitelist) {
+    if (!whitelist.isEmpty()) {
+      return stringInPatterns(topic, whitelist);
+    } else {
+      return !stringInPatterns(topic, blacklist);
+    }
+  }
+
+  private static boolean stringInPatterns(String s, List<Pattern> patterns) {
+    for (Pattern pattern : patterns) {
+      if (pattern.matcher(s).matches()) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private void processTopics(Set<String> topics) throws IOException {
+    ProcessHighPriorityTopics(topics);
+    ProcessNormalPriorityTopics(topics);
+    ProcessRemainingTopics(topics);
+  }
+
+  private void ProcessHighPriorityTopics(Set<String> topics) throws IOException {
```

Comment:
lower-case P at the beginning of method name. Same for normal and remaining topics.